### PR TITLE
[tests-only] Added more GUI tests for sharing with user (part 02)

### PR DESCRIPTION
**Pull request by saw-jan**
### Description
Added more automated tests for different sharing scenarios

Splitted from https://github.com/owncloud/client/pull/9323

### CODE REVIEW

File: test/gui/shared/scripts/bdd_hooks.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def hook(context):
     DEFAULT_CONFIG = {
         'localBackendUrl': 'https://localhost:9200/',
         'secureLocalBackendUrl': 'https://localhost:9200/',
-        'clientSyncTimeout': 60,
+        'clientSyncTimeout': 10,
```
Comment (Talank) [inline test/gui/shared/scripts/bdd_hooks.py:40]: This PR seems to be dependent upon changes made in some other PRs (part 1 and 3). Does the order of merge should be Part1 -> part2 -> part3 ?

Comment (saw-jan) [reply test/gui/shared/scripts/bdd_hooks.py:40]: the actual order should not affect each other. one needs to be rebased. So whichever gets merged first, it should not create affect other PR

**Review by Talank — COMMENTED** [review]
lgmt

File: test/gui/tst_sharing/test.feature [hunk 2]
```diff
@@ -98,44 +143,112 @@ Feature: Sharing
...
+        And as "Alice" the file "textfile.txt" on the server should have the content "overwrite ownCloud test text file"
+        And as "Brian" the file "FOLDER/simple.txt" on the server should have the content "overwrite some content"
+        And as "Alice" the file "FOLDER/simple.txt" on the server should have the content "overwrite some content"
+        When the user adds another account with
+            | server   | %local_server% |
+            | user     | Alice          |
+            | password | 1234           |
+        And the user removes permissions "edit" for user "Brian Murphy" of resource "textfile.txt" using the client-UI
+        And the user closes the sharing dialog
+        And the user removes permissions "edit" for user "Brian Murphy" of resource "FOLDER" using the client-UI
```
Comment (individual-it) [inline test/gui/tst_sharing/test.feature:245]: is this test about changing the sharing from the UI or about the overwriting?
Maybe we could have those steps done by the API?

File: test/gui/shared/steps/steps.py [hunk 3]
```diff
@@ -920,9 +919,28 @@ def step(context, resource, group):
...
+
+
+def tryToOverwriteFile(context, resource, content):
+    waitForFileToBeSynced(context, resource)
+    waitForFolderToBeSynced(context, '/')
+
+    try:
+        overwriteFile(resource, content)
+    except:
+        pass
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:935]: why would that ever fail? Does the client set read-only permissions on the local file?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/tst_sharing/test.feature (hunk 2 above)
Comment (saw-jan) [reply test/gui/tst_sharing/test.feature:245]: The scenario is to do `overwrite` -> `change permissions` -> `try to overwrite` using the client UI.
Any suggestions?

File: test/gui/shared/steps/steps.py (hunk 3 above)
Comment (saw-jan) [reply test/gui/shared/steps/steps.py:935]: I found out that when a folder is shared with `read` permission, files inside it have `-r--r--r--` permissions but the folder itself has `drwxrwxr-x` permissions in the files system.
And when a file is shared with `read` permission, the file has `-rw-rw-r--` permissions

Comment (individual-it) [reply test/gui/shared/steps/steps.py:935]: same happens when you do it manually, or only in the test?

Comment (saw-jan) [reply test/gui/shared/steps/steps.py:935]: Yeah, it is the same but only if we share-edit-permissions first from the server and then add the user to the client.

**Review by individual-it — APPROVED** [review]

**Pull request merged**